File: and/ttp/2018/semana1/contest/3644.cpp
#include <stdio.h>
#include <cmath>

int main() {
    long long n;
    while (scanf("%lld", &n) != EOF) {
        long long cnt = 0;
        while (n) {
            long long f = floor((sqrt(1.0 + 8 * n) - 1) / 2.0);
            n -= (f * (f + 1)) / 2;
            cnt++;
        }
        printf("%lld\n", cnt);
    }
}

File: and/ttp/2018/semana1/contest/3644.py
#!/usr/bin/env python2
import math

while True:
    n = 0
    try:
        n = input()
    except:
        break
    cnt = 0
    while n>0:
        f = math.floor((-1+math.sqrt(1+8*n))/2)
        n-=(f*(f+1))/2
        cnt+=1
    print cnt